### PR TITLE
Fix DeterministicOutput, add tests

### PR DESCRIPTION
**Pull request by lostella**
*Description of changes:* Fixes the `DeterministicOutput` class and add tests for it. This class is useful e.g. to implement inflated distributions (through a `MixtureDistributionOutput`).

cc @PascalIversen 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/mx/distribution/distribution_output.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def dtype(self):
     def dtype(self, dtype: DType):
         self._dtype = dtype
 
-    def get_args_proj(self, prefix: Optional[str] = None) -> ArgProj:
+    def get_args_proj(self, prefix: Optional[str] = None) -> gluon.HybridBlock:
```
Comment (lostella) [inline src/gluonts/mx/distribution/distribution_output.py:96]: This annotation seems to me more appropriate

File: src/gluonts/mx/distribution/deterministic.py [hunk 2]
```diff
@@ -88,7 +89,7 @@ def cdf(self, x):
         return is_greater_equal_or_both_nan
 
     def sample(
-        self, num_samples: Optional[int] = None, dtype=np.int32
```
Comment (lostella) [inline src/gluonts/mx/distribution/deterministic.py:91]: @PascalIversen was there any reason for this to be `np.int32`?

Comment (PascalIversen) [reply src/gluonts/mx/distribution/deterministic.py:91]: No. Good catch, thanks!

**Comment by lostella** [comment]
> because the deterministic distribution returns the log probability and not the log density (which would be infinite).

right, there's a bit of an abuse of notation in all distribution types

**Pull request merged**